paddle/operators/trieconcat_op.h

```diff
@@ -0,0 +1,248 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserve.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#pragma once
+
+#include "paddle/framework/op_registry.h"
+
+namespace paddle {
+namespace operators {
+
+using value_type = float;
+using LoDTensor = framework::LoDTensor;
+
+const int64_t kInitLength = 1024;
+const int64_t kEndId = 0;
+
+// all the lod have 2 level, the first it source level,
+// each source have multiple possible sentences in the second level
+const size_t kSourceLevel = 0;
+const size_t kSentenceLevel = 1;
+
+struct BeamNode {
+  BeamNode(int64_t word_id, float prob) : word_id_(word_id), prob_(prob) {}
```

Comment:
You need a destructor here

```cpp
~BeamNode() {
  if (parent_) {
     parent_->DropKid(this);
     if (parent_->NumKid() == 0) {
       delete parent_;
     }
  }
}
```

Reply:
currently, I use a function [RemoveFromEnd](https://github.com/PaddlePaddle/Paddle/pull/5498/files#diff-05b068ed70104191c900a42615b154b8R49) to do this, but it need to be optimized~ thank you for your suggestion.

Reply:
done